From bdc1aec70aab30b075558b98657bac55d4845963 Mon Sep 17 00:00:00 2001
From: Vishesh Handa <me@vhanda.in>
Date: Tue, 31 Jul 2012 22:27:16 +0530
Subject: [PATCH 01/10] No qDebug output in non debug modes

Soprano should not emit debug output when compiling with Release or
RelWithDebInfo.

BUG: 247483
REVIEW: 105807
---
 CMakeLists.txt | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7fe4d73..76e4962 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -45,6 +45,14 @@ include(SopranoCPack.cmake)
 set(QT_MIN_VERSION "4.5.0")
 find_package(Qt4 REQUIRED)
 # properly set up compile flags (QT_DEBUG/QT_NO_DEBUG, ...)
+if(CMAKE_BUILD_TYPE)
+    string(TOLOWER ${CMAKE_BUILD_TYPE} CMAKE_BUILD_TYPE_TOLOWER)
+endif()
+
+if (NOT CMAKE_BUILD_TYPE_TOLOWER MATCHES "debug") 
+   add_definitions(-DQT_NO_DEBUG_OUTPUT) 
+endif()
+
 include(${QT_USE_FILE})
 
 if(NOT SOPRANO_DISABLE_RAPTOR_PARSER OR NOT SOPRANO_DISABLE_RAPTOR_SERIALIZER OR NOT SOPRANO_DISABLE_REDLAND_BACKEND)
-- 
1.8.0.2

